File: src/lessons/welcome/array/occurenceofvalue/OccurrenceOfValueEntity.java
package lessons.welcome.array.occurenceofvalue;

import jlm.universe.array.ArrayEntity;

public class OccurrenceOfValueEntity extends ArrayEntity {

	@Override
	public void run() {
		setResult( occurrences(getValues(), 17) );
	}

	/* BEGIN TEMPLATE */
	// computes the occurrences of the value 'lookingFor' contained in tab variable
	public int occurrences(int[] tab, int lookingFor) {
		/* BEGIN SOLUTION */
		int count = 0;
		for (int i=0; i<tab.length; i++) {
			if (tab[i] == lookingFor) {
				count++;
			}
		}
		return count;
		/* END SOLUTION */
	}

	/* END TEMPLATE */

}

File: src/lessons/welcome/array/occurenceofvalue/OccurrenceOfValueEntity.py
# BEGIN TEMPLATE
# Counts the occurrences of the value 'lookingFor' contained in tab variable
def indexOf(tab,lookingFor):
	# BEGIN SOLUTION
	count = 0
	for i in range(len(tab)):
		if tab[i] == lookingFor:
			count += 1
	return count
# END SOLUTION
# END TEMPLATE

setResult( indexOf(getValues(), 17) )
